### PR TITLE
[WIP] Catalog server local setup

### PR DESCRIPTION
**Pull request by zacw7**
Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
* ...
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @ThriftMethod
+    public boolean catalogExists(String catalogName)
+    {
+        return catalogManager.getCatalog(catalogName).isPresent();
+    }
+
+    @ThriftMethod
+    public boolean schemaExists(SessionRepresentation session, String catalogName, String schemaName)
```
Comment (yuanzhanhku) [inline presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java:59]: The session can contain lots of values. Passing the entire session for each CatalogServer call may be too heavy.

File: presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        Optional<Catalog> catalog = getCatalog(catalogName);
+        if (!catalog.isPresent()) {
```
Comment (yuanzhanhku) [inline presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java:62]: Inlining the implementation in the Thrift Server implementation will result in two copies of the code.
We can have an interface for local and remote MetadataManager. The remote MetadataManager should send Thrift request to the CatalogServer which just invokes the function of a local MetadataManager. In this case there is only implementation in the local MetadataManager.

Comment (yuanzhanhku) [inline presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java:62]: Inlining the implementation in the Thrift Server implementation will result in two copies of the code.
We can have an interface for local and remote MetadataManager. The remote MetadataManager should send Thrift request to the CatalogServer which just invokes the function of a local MetadataManager. In this case there is only implementation in the local MetadataManager.

File: presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java [hunk 3]
```diff
@@ -165,9 +165,7 @@ public void run()
             injector.getInstance(StaticFunctionNamespaceStore.class).loadFunctionNamespaceManagers();
             injector.getInstance(SessionPropertyDefaults.class).loadConfigurationManager();
             injector.getInstance(ResourceGroupManager.class).loadConfigurationManager();
-            if (!serverConfig.isResourceManager()) {
```
Comment (yuanzhanhku) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java:168]: Why do we need to make this change?

Comment (yuanzhanhku) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java:168]: Why do we need to make this change?

Comment (zacw7) [reply presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java]: Seems not needed anymore

File: presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java (hunk 1 above)
Comment (zacw7) [reply presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java:59]: Not sure if we can avoid session. It is required by the lowest level [ConnectorMetadata](https://github.com/prestodb/presto/blob/master/presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java).

**Pull request closed without merging**